### ISSUE TITLE
Polygon type?

### ISSUE THREAD

**Issue by goretkin** [JuliaGeometry/GeometryTypes.jl#124]
Should there be an explicit Polygon type? Right now the code uses `AbstractArray{Point}` and `Vector{Point}` (in only one place each) to describe a Polygon.

**Comment by SimonDanisch** [comment]
Yes, that would be appreciated! We had some discussions about this, but never came to a conclusion, since people seem to have very different needs for polygons. I was thinking to make `Vector{Vector{Point}}` our first polygon type for now, since that's what e.g. https://github.com/JuliaGeometry/EarCut.jl expects right now, and I wish to integrate that better into the geometry infrastructure! But feel free to go ahead with any polygon type you want! We are pretty undecided on the exact definition, so it's a bit first come, first serve kind of situation ;)
